hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/CompatibilitySingletonFactory.java

```diff
@@ -31,6 +31,8 @@
  */
 @InterfaceAudience.Private
 public class CompatibilitySingletonFactory extends CompatibilityFactory {
+
+  @SuppressWarnings("ImmutableEnumChecker")
```

Comment:
Not a good design but anyway, it works. Not your fault, let's file another issue to rewrite this...